tests/mock_virtual_machine.h

```diff
@@ -33,7 +33,8 @@ namespace test
 template <typename T = VirtualMachine, typename = std::enable_if_t<std::is_base_of_v<VirtualMachine, T>>>
 struct MockVirtualMachineT : public T
 {
-    MockVirtualMachineT(const std::string vm_name) : T{vm_name}
+    template <typename... Args>
+    MockVirtualMachineT(Args&&... args) : T{args...}
```

Comment:
Nice, this was missing indeed. But may as well use perfect forwarding:

```suggestion
    template <typename... Args>
    MockVirtualMachineT(Args&&... args) : T{std::forward<Args>(args)...}
```

Reply:
Yep, you are right, can not believe I have overlooked that. Universal reference `Args&& `does retain value type in general. However, when rvalue `args` get into the function (`MockVirtualMachineT` constructor in this case), it gets a name, a variable with a name becomes lvalue. So the `args` passed to sub-function (`T` constructor in this case) without `std::forward` will be lvalue instead of rvalue, which might cause either wrong constructor match or can not find the wanted constructor.